src/hotspot/share/memory/metaspace/metachunk.cpp

```diff
@@ -108,7 +170,28 @@ bool Metachunk::commit_up_to(size_t new_committed_words) {
 
   // Remember how far we have committed.
   _committed_words = commit_to;
+
+  // If this chunk was smaller than a granule we just committed, we committed the memory
+  //  underlying the in-granule neighbors as well. Lets correct their commit boundaries.
+  // Note that this is not strictly necessary: these boundaries would silently be corrected
+  //  the first time someone were to use those chunks. But doing it now keeps statistics happy
+  //  and we save some work later on.
+  if (word_size() < Settings::commit_granule_words()) {
+    const MetaWord* granule_start = align_down(base(), Settings::commit_granule_bytes());
+    const MetaWord* granule_end = granule_start + Settings::commit_granule_words();
+    assert(granule_end >= end(), "Sanity");
+    for (Metachunk* c = prev_in_vs(); c != NULL && c->base() >= granule_start; c = c->prev_in_vs()) {
+      assert(c->committed_words() == 0, "neighbor was already committed?");
+      c->_committed_words = c->word_size();
```

Comment:
Can the previous metachunk be partially contained in the commit granule?  ie c->base < granule_start but c->end >= granule start?

Reply:
No. A chunk smaller than a granule is fully contained in one granule; a chunk larger than a granule contains only full granules.

Which is inbuilt since both chunks and granules are pow2 values, and start at addresses aligned to their size.

But just to make things very explicit I'll add an assert for this.